scripts/update-pkgs-2021.12.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -e
+
+if [ -z "$EPREFIX" ]; then
+    # this assumes we're running in a Gentoo Prefix environment
+    EPREFIX=$(dirname $(dirname $SHELL))
+fi
+
+# first update checkout of gentoo repository to sufficiently recent commit
+# this is required because we pin to a specific commit when bootstrapping the compat layer
+# see gentoo_git_commit in ansible/playbooks/roles/compatibility_layer/defaults/main.yml
+cd $EPREFIX/var/db/repos/gentoo
+git fetch origin
+git checkout d7c647404a632309810851c52c1d350cafc26949  # 2022-08-25
+cd -
+
+# unmask GCC 9.4, since we're using that as system compiler in EESSI pilot 2021.12;
+# see also $EPREFIX/var/db/repos/gentoo/profiles/package.mask
+echo '=sys-devel/gcc-9.4.0' >> ${EPREFIX}/etc/portage/package.unmask
+
+# stick to Python 3.9
+# see also https://wiki.gentoo.org/wiki/Project:Python/PYTHON_TARGETS
+# note: make sure that ${EPREFIX}/etc/python-exec/python-exec.conf has the right version of Python in it too
+echo '# replace profile default Python with Python 3.9' >> ${EPREFIX}/etc/portage/package.use
+echo '*/* PYTHON_TARGETS: -* python3_9' >> ${EPREFIX}/etc/portage/package.use
+echo '*/* PYTHON_SINGLE_TARGET: -python3_10 python3_9' >> ${EPREFIX}/etc/portage/package.use
+
+emerge --sync
+
+# update lxml due to https://glsa.gentoo.org/glsa/202208-06
+# requires commit b93e546746f6b60c5efe97ad6d98636547c3b797 (2022-07-02)
+emerge --oneshot --verbose '=dev-python/lxml-4.9.1'
```

Comment:
Running this script again in a Prefix environment where these packages have already been updated will install them again...

@amadio Is there an easy way to avoid that, so the script it idempotent when it's run multiple times?

Reply:
Add `-u` to emerge's options.

Reply:
Thanks, works like a charm:

```
These are the packages that would be merged, in order:

Calculating dependencies... done!

Total: 0 packages, Size of downloads: 0 KiB

Nothing to merge; quitting.
```